### ISSUE TITLE
📎 Port `no-extend-native` from ESLint

### ISSUE THREAD

**Issue by esaari** [biomejs/biome#10908]
### Description

Please consider porting the ESLint core rule [`no-extend-native`](https://eslint.org/docs/latest/rules/no-extend-native).

Extending the prototype of a built-in (`Object`, `Array`, `Error`, …) is a well-known footgun: it leaks enumerable properties into every `for…in`, collides with other libraries, and breaks assumptions across the whole program. It's a bug-class check rather than a style preference.

**Invalid**
```js
Object.prototype.a = "a";
Array.prototype.times = function () {};
Object.defineProperty(Array.prototype, "times", { value: 999 });
```

**Valid**
```js
class CustomArray extends Array {}
const obj = {};
obj.a = "a";
```

- Flags: a direct prototype assignment (`<Builtin>.prototype.x = …`) and `Object.defineProperty` / `Object.defineProperties` targeting a built-in prototype.
- Built-ins: the standard constructor globals (`Object`, `Function`, `Array`, `Number`, `Boolean`, `String`, `Symbol`, `BigInt`, `Date`, `RegExp`, `Error` + subclasses, `Map`, `Set`, `WeakMap`, `WeakSet`, `Promise`, `ArrayBuffer`, `DataView`, the typed arrays).
- Not auto-fixable (removing a native extension isn't a mechanical rewrite) — detection-only, matching the source rule.

Suggested name/category: `noExtendNative` under **suspicious**, happy to defer to your conventions.

I'd like to implement it (I'm also working on #10899 / #10907).

### Source
- ESLint core / `no-extend-native`

**Comment by dyc3** [comment]
Yeah, this would also be a good rule to have.

**Comment by esaari** [comment]
Opened #10909 with the rule (detection-only: direct + computed prototype assignment and `Object.defineProperty`/`defineProperties` on a native prototype).

**Comment by AkashNaickar** [comment]
Opened PR #11136 with the implementation.
